testing/test_mercurial.py

```diff
@@ -32,7 +32,12 @@ def wd(wd: WorkDir) -> WorkDir:
 archival_mapping = {
     "1.0": {"tag": "1.0"},
     "1.1.0.dev3+h000000000000": {
-        "latesttag": "1.0",
+        "latesttag": "1.0.0",
+        "latesttagdistance": "3",
```

Comment:
i need some more examples, i cant quite wrap my head around it

this is a orkaround to ease the pain, but its also unable to completely prevent it